### PR TITLE
verify ristretto::decompress

### LINKED ISSUE #494: Verify `decompress::step_1` in `edwards.rs`

### PR DESCRIPTION
**Pull request by astefano**
- [ ] Source code modifications **highlighted and justified**
- [ ] Proof cheats (`assume(false)`, `sorry`, etc.) **highlighted and justified**

Verify the Ristretto point decompression function (`CompressedRistretto::decompress`), eliminating all `assume()` / `assume(false)` calls from `step_1`, `step_2`, and the top-level `decompress` wrapper. The proof establishes coordinate-level correctness against a new `spec_ristretto_decompress_coords` spec, well-formedness of the output `EdwardsPoint`, and membership in the even subgroup.

## Specs changed

- **`ristretto_specs.rs`**
  - Added `spec_ristretto_decompress_coords` — a coordinate-level decompression spec returning `Option<(nat, nat, nat, nat)>` that avoids the unsound `choose|p: RistrettoPoint|` in the existing `spec_ristretto_decompress`. This is the spec that `decompress` now proves against.
  - Added `spec_ristretto_decode_inner(s) -> (nat, nat, bool)` — factors out the shared decode computation (u1, u2, v, invsqrt, Dx, Dy, x, y, ok) so both specs and axioms reference a single formula.
  - Added `spec_ristretto_decode_x`, `spec_ristretto_decode_y`, `spec_ristretto_decode_ok` — projections from `spec_ristretto_decode_inner`.
  - Added `is_ristretto_decode_output(s, I, x, y)` — predicate for axiom preconditions, asserting the Ristretto decode relationships hold between s, the inverse square root I, and output coordinates x, y.
  - Refactored `spec_ristretto_decompress` to delegate to `spec_ristretto_decode_inner` (no semantic change) and added a deprecation note about the `choose` issue.
- **`field_specs.rs`**
  - Strengthened `nat_invsqrt` choose predicate: added `r < p()` to ensure the chosen value is the canonical representative, making `axiom_invsqrt_unique` sound.

## Axioms introduced (with `admit()`)

| Axiom | File | Purpose | Runtime test |
|---|---|---|---|
| `axiom_ristretto_decode_on_curve` | `ristretto_specs.rs` | Decoded (x, y) satisfies the Edwards curve equation (restricted to the square case where I²·v·u2² = 1) | `test_ristretto_decode_on_curve` |
| `axiom_ristretto_decode_in_even_subgroup` | `ristretto_specs.rs` | Decoded point lies in the even subgroup 2E | `test_ristretto_decode_in_even_subgroup` |
| `axiom_invsqrt_unique` | `sqrt_ratio_lemmas.rs` | Nonneg inverse square root is uniquely determined | `test_invsqrt_unique` |
| `axiom_invsqrt_exists` | `sqrt_ratio_lemmas.rs` | A nonneg inverse square root exists for every nonzero field element (justifies the `choose` in `nat_invsqrt`) | `test_invsqrt_unique` |

## Axioms eliminated / proven

- **`axiom_sqrt_m1_limbs_bounded`** → renamed to `lemma_sqrt_m1_limbs_bounded` and fully proven via concrete limb value assertions + `bit_vector` mode.
- **`axiom_sqrt_ratio_mutual_exclusion`** → fully proven as `lemma_sqrt_ratio_mutual_exclusion` using `lemma_no_square_root_when_times_i`.
- **`axiom_ristretto_point_from_coords`** — removed (was unsound: `spec_edwards_point` maps through `fe51_as_canonical_nat` which is not injective on limb representations).

## Postcondition changes

`decompress` previously had:
```
result == spec_ristretto_decompress(self.0)
```
This is replaced with:
```
result.is_none() <==> spec_ristretto_decompress_coords(self.0).is_none(),
result.is_some() ==> spec_edwards_point(result.unwrap().0)
    == spec_ristretto_decompress_coords(self.0).unwrap(),
result.is_some() ==> is_well_formed_edwards_point(result.unwrap().0),
result.is_some() ==> is_in_even_subgroup(result.unwrap().0),
```

## Runtime tests added

- `test_ristretto_decode_on_curve` — tests on-curve axiom across basepoint, multiples, edge cases, and 50 hash-derived inputs
- `test_ristretto_decode_in_even_subgroup` — tests [L]P = identity for decoded points (basepoint, 48 multiples, 8 diverse inputs, 50 hash-derived)
- `test_invsqrt_unique` — tests uniqueness + existence for 190+ field elements including non-squares
- `test_sqrt_m1_limbs_bounded` — tests each limb of SQRT_M1 fits in 51 bits, plus i² = -1 sanity check
- `test_minus_one_field_element_value` — tests MINUS_ONE constant
- `test_sqrt_m1_not_square` — tests i and -i are non-squares via Euler's criterion
- `test_p_is_prime` — tests primality of p = 2²⁵⁵ - 19 via Miller-Rabin with 9 witnesses + Fermat check

## Documentation

- Added `docs/proofs_for_ristretto_decompress/` with mathematical proofs (`decompress.md`, `step_1.md`, `step_2.md`) and a Verus proof mapping (`verus_proof_mapping.md`).
- Added `docs/axiom_references.md` documenting all axioms in the codebase.

closes #494, #430, #428

### CODE REVIEW

File: curve25519-dalek/src/ristretto.rs [hunk 1]
```diff
@@ -434,68 +544,385 @@ mod decompress {
...
+            }
+        }
+
+        // Construct EdwardsPoint: use computed values when ok (proven on-curve),
+        // use identity() when !ok to avoid unsound assume in degenerate case.
+        let point = if choice_into(ok) {
+            EdwardsPoint { X: x, Y: y, Z: one, T: t }
+        } else {
+            EdwardsPoint::identity()
+        };
```
[the remark this answers is not in the document]
Comment (sergiubur) [reply curve25519-dalek/src/ristretto.rs]: problem is that the original code breaks the type invariant

[the remark this answers is not in the document]
Comment (sergiubur) [reply curve25519-dalek/src/ristretto.rs]: ✅

[the remark this answers is not in the document]
Comment (sergiubur) [reply curve25519-dalek/src/ristretto.rs]: Signalled this on: https://github.com/dalek-cryptography/curve25519-dalek/discussions/889
